control-plane/agents/src/bin/core/volume/snapshot_helpers.rs

```diff
@@ -95,34 +95,32 @@ impl SpecOperationsHelper for VolumeSnapshot {
 pub(crate) async fn snapshoteable_replica(
     volume: &VolumeSpec,
     registry: &Registry,
-) -> Result<ChildItem, SvcError> {
-    if volume.num_replicas != 1 {
-        return Err(SvcError::NReplSnapshotNotAllowed {});
-    }
-
+) -> Result<Vec<ChildItem>, SvcError> {
     let children = super::scheduling::snapshoteable_replica(volume, registry).await?;
 
     volume.trace(&format!("Snapshoteable replicas for volume: {children:?}"));
 
-    let item = match children.candidates().as_slice() {
-        [item] => Ok(item),
-        [] => Err(SvcError::NoHealthyReplicas {
-            id: volume.uuid_str(),
-        }),
-        _ => Err(SvcError::NReplSnapshotNotAllowed {}),
-    }?;
+    if children.candidates().len() != volume.num_replicas as usize
+        || children.candidates().is_empty()
+    {
+        return Err(SvcError::AllReplicaNotHealthy {});
+    }
```

Comment:
Separate the errors out

```suggestion
    if children.candidates().is_empty() {
        return Err(SvcError::NoHealthyReplicas { id: volume.uuid_str() });
    }
    if children.candidates().len() != volume.num_replicas as usize {
        return Err(SvcError::AllReplicaNotHealthy { id: volume.uuid_str() });
    }
```